data-prepper-plugins/aggregate-processor/src/main/java/com/amazon/dataprepper/plugins/processor/aggregate/AggregateAction.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.amazon.dataprepper.plugins.processor.aggregate;
+
+import com.amazon.dataprepper.model.event.Event;
+
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * Interface for creating custom actions to be used with the {@link com.amazon.dataprepper.plugins.processor.aggregate.AggregateProcessor}.
+ * @since 1.3
+ */
+public interface AggregateAction {
+    /**
+     * Handles an event as part of aggregation.
+     *
+     * @param event The current event
+     * @param groupState An arbitrary map for the current group
```

Comment:
AggregationAction authors are going to rely on the documentation in these Javadocs. So I'd like to elaborate on the groupState some. Here is one possible sentence, but I'm open to others than convey the important points.

> A map for AggregationActions to maintain user-defined for a group. The same map is provided for all events in a single group. This map is non-null.